python/ray/_private/node.py

```diff
@@ -1238,6 +1235,9 @@ def start_head_processes(self):
         else:
             raise_on_api_server_failure = self._ray_params.include_dashboard
 
+        assert self.get_gcs_client() is not None
+        self._write_cluster_info_to_kv()
```

Comment:
I think it's fragile: there is a hidden assumption here that monitor and ray client server don't need to access cluster info populated by `_write_cluster_info_to_kv`

Reply:
So I guess a more enduring fix might look something like: let GCS serve the kv store before the rest of the server has fully initialized, but there's the whole problem of GCS being single-threaded

Reply:
passing the metadata to the processes when they're started is not the best solution, but it's /a/ solution